xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
Don't mix up things. We deal with:
1) a/v sync
Presentation timestamp of a video frame relates to the time the frame becomes visible. In order to sync audio to this time we need the offset beween vertical sync signal and video pts.

2) queue video frames into GL.
When the render threads comes by we need to decide if the next frame in our render buffers is due. This is not completely trival because video drivers may queue frames internally. When playing some 50 fps content for example, the render thread can be blocked close to 40ms and we don't necessarily lose a frame. The next two cycles will be done in much less than frametime. RM needs to account for this elasticity. Moving the decision point to t(vsync) + frametime / 2 makes it most reliable.